### PR TITLE
instrumentation/python: set OTEL_EXPORTER_OTLP_PROTOCOL to http/protobuf

### PR DESCRIPTION
**Pull request by xrmx**
**Description:** <Describe what has changed.>

Instead of setting the environment variable for each specific signal use the common one that it is also the one that is set by the distro.

**Link to tracking Issue(s):** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

Added test case for the new environment variable set

**Documentation:** <Describe the documentation added.>

None

### CODE REVIEW

**Comment by xrmx** [comment]
I'm not able to sort out what's wrong with the e2e-tests failure but looks unrelated.

**Comment by jaronoff97** [comment]
@xrmx fixed that, e2e can be flaky at times.

**Review by jaronoff97 — APPROVED** [review]

File: pkg/instrumentation/python.go [hunk 1]
```diff
@@ -71,6 +72,15 @@ func injectPythonSDK(pythonSpec v1alpha1.Python, pod corev1.Pod, index int) (cor
 		})
 	}
 
+	// Set OTEL_EXPORTER_OTLP_PROTOCOL to http/protobuf if not set by user because it is what our autoinstrumentation supports.
+	idx = getIndexOfEnv(container.Env, envOtelExporterOTLPProtocol)
```
Comment (TylerHelmuth) [inline pkg/instrumentation/python.go:76]: If we set the top-level env var we no longer need to set the signal specific-env vars.

Comment (xrmx) [reply pkg/instrumentation/python.go:76]: Yep, will drop signal specific env vars

**Review by TylerHelmuth — APPROVED** [review]

File: pkg/instrumentation/python.go (hunk 1 above)
Comment (xrmx) [reply pkg/instrumentation/python.go:65]: Updated PR

**Comment by xrmx** [comment]
I don't understand what this means:

```
pkg/instrumentation/python.go:26: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/open-telemetry/opentelemetry-operator) (gci)
```

**Comment by TylerHelmuth** [comment]
@xrmx it means the import statements at the top of the file are not arranged correctly into 3 sections.

**Comment by xrmx** [comment]
> @xrmx it means the import statements at the top of the file are not arranged correctly into 3 sections.

Was not that, the issue was that there were too many spaces in the const definitions before the `"` :sweat_smile:

**Review by iblancasa — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

File: .chloggen/3165-python-otlp-protocol-env-var.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: collector
```
Comment (pavolloffay) [inline .chloggen/3165-python-otlp-protocol-env-var.yaml:5]: this should be set to auto-instrumentation

Comment (xrmx) [inline .chloggen/3165-python-otlp-protocol-env-var.yaml:5]:
```suggestion
component: auto-instrumentation
```

Comment (xrmx) [reply .chloggen/3165-python-otlp-protocol-env-var.yaml]: Updated, thanks

**Comment by xrmx** [comment]
Hello, can I do something to move this forward? BTW merged main and e2e tests failure looks unrelated. Thanks!

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**